lib/Scheduling/TestPasses.cpp

```diff
@@ -83,13 +83,11 @@ static void constructProblem(Problem &prob, FuncOp func) {
   // of integer attributes
   if (auto attr = func->getAttrOfType<ArrayAttr>("auxdeps")) {
     auto &ops = prob.getOperations();
-    unsigned nOps = ops.size();
-    for (auto &elemArr : parseArrayOfArrays(attr)) {
+    [[maybe_unused]] unsigned nOps = ops.size();
+    for ([[maybe_unused]] auto &elemArr : parseArrayOfArrays(attr)) {
```

Comment:
This is really ugly...

Reply:
You can wrap the entire for loop in `#ifndef NDEBUG`, e.g. https://github.com/llvm/llvm-project/blob/main/mlir/lib/Dialect/Linalg/ComprehensiveBufferize/ModuleBufferization.cpp#L218

Reply:
Thanks for reviewing. late to reply. 
**#ifndef NDEBUG** is common used in llvm, will try with it. 
Although  find **[[maybe_unused]]** in https://github.com/llvm/llvm-project/blob/main/flang/lib/Optimizer/Transforms/MemoryAllocation.cpp#L126 ,
may consider not use it.